blueflood-core/src/main/java/com/rackspacecloud/blueflood/inputs/processors/DiscoveryWriter.java

```diff
@@ -91,7 +138,11 @@ private static List<IMetric> condense(List<List<IMetric>> input) {
             }
 
             for (IMetric m : list) {
-                if (!LocatorCache.getInstance().isLocatorCurrentInDiscoveryLayer(m.getLocator())) {
+                boolean isAlreadySeen = LocatorCache.getInstance().isLocatorCurrentInDiscoveryLayer(m.getLocator());
+                boolean isThrottlingGlobally = newLocatorsThrottle.size() >= maxNewLocatorsPerMinute;
```

Comment:
global throttle check can be done once for all metrics, isn't it?

Reply:
You're right. Good catch. Fixed in #913